tests/braille-specs/tr.yaml

```diff
@@ -0,0 +1,870 @@
+  - ["Ż", ⠠⠈⠵, {xfail: {backward: conflict}}]
+  - ["Ž", ⠠⠈⠵, {xfail: {backward: conflict}}]
+  - ["Ǎ", ⠠⠈⠁, {xfail: {backward: conflict}}]
+  - ["Ẁ", ⠠⠈⠺, {xfail: {backward: conflict}}]
+  - ["Ẃ", ⠠⠈⠺, {xfail: {backward: conflict}}]
+  - ["Ẅ", ⠠⠈⠺, {xfail: {backward: conflict}}]
+  - ["Ỳ", ⠠⠈⠽, {xfail: {backward: conflict}}]
+  - ["ț", ⠈⠞, {xfail: {backward: conflict}}]
+  - ["ũ", ⠈⠥, {xfail: {backward: conflict}}]
+  - ["ű", ⠈⠥, {xfail: {backward: conflict}}]
+  - ["ų", ⠈⠥, {xfail: {backward: conflict}}]
+  - ["ŵ", ⠈⠺, {xfail: {backward: conflict}}]
+  - ["ŷ", ⠈⠽, {xfail: {backward: conflict}}]
+  - ["ź", ⠈⠵]
+  - ["ż", ⠈⠵, {xfail: {backward: conflict}}]
+  - ["ž", ⠈⠵, {xfail: {backward: conflict}}]
+  - ["ǎ", ⠈⠁, {xfail: {backward: conflict}}]
+  - ["ẁ", ⠈⠺, {xfail: {backward: conflict}}]
+  - ["ẃ", ⠈⠺, {xfail: {backward: conflict}}]
+  - ["ẅ", ⠈⠺, {xfail: {backward: conflict}}]
+  - ["ỳ", ⠈⠽, {xfail: {backward: conflict}}]
+  - ["À", ⠠⠈⠁, {xfail: {backward: conflict}}]
+  - ["Á", ⠠⠈⠁, {xfail: {backward: conflict}}]
+  - ["Â", ⠠⠈⠁]
+  - ["Ã", ⠠⠈⠁, {xfail: {backward: conflict}}]
+  - ["Ä", ⠠⠈⠁, {xfail: {backward: conflict}}]
+  - ["Å", ⠠⠈⠁, {xfail: {backward: conflict}}]
+  - ["Æ", ⠠⠠⠁⠑, {xfail: {backward: true}}]
+  - ["Ç", ⠠⠡]
+  - ["È", ⠠⠈⠑]
+  - ["É", ⠠⠈⠑, {xfail: {backward: conflict}}]
+  - ["Ê", ⠠⠈⠑, {xfail: {backward: conflict}}]
+  - ["Ë", ⠠⠈⠑, {xfail: {backward: conflict}}]
+  - ["Ì", ⠠⠈⠊, {xfail: {backward: conflict}}]
+  - ["Í", ⠠⠈⠊, {xfail: {backward: conflict}}]
+  - ["Î", ⠠⠈⠊]
+  - ["Ï", ⠠⠈⠊, {xfail: {backward: conflict}}]
+  - ["Ð", ⠠⠈⠙]
+  - ["Ñ", ⠠⠈⠝]
+  - ["Ò", ⠠⠈⠕]
+  - ["Ó", ⠠⠈⠕, {xfail: {backward: conflict}}]
+  - ["Ô", ⠠⠈⠕, {xfail: {backward: conflict}}]
+  - ["Õ", ⠠⠈⠕, {xfail: {backward: conflict}}]
+  - ["Ö", ⠠⠪]
+  - ["Ø", ⠠⠈⠕, {xfail: {backward: conflict}}]
+  - ["Ù", ⠠⠈⠥, {xfail: {backward: conflict}}]
+  - ["Ú", ⠠⠈⠥, {xfail: {backward: conflict}}]
+  - ["Û", ⠠⠈⠥]
+  - ["Ü", ⠠⠳]
+  - ["Ý", ⠠⠈⠽]
+  - ["Þ", ⠠⠈⠞, {xfail: {backward: conflict}}]
+  - ["Ÿ", ⠠⠈⠽, {xfail: {backward: conflict}}]
+  - ["Œ", ⠠⠠⠕⠑, {xfail: {backward: conflict}}]
+  - ["Ā", ⠠⠈⠁, {xfail: {backward: conflict}}]
+  - ["Ă", ⠠⠈⠁, {xfail: {backward: conflict}}]
+  - ["ă", ⠈⠁, {xfail: {backward: conflict}}]
+  - ["Ą", ⠠⠈⠁, {xfail: {backward: conflict}}]
+  - ["ą", ⠈⠁, {xfail: {backward: conflict}}]
+  - ["Ć", ⠠⠈⠉]
+  - ["ć", ⠈⠉]
+  - ["Ĉ", ⠠⠈⠉, {xfail: {backward: conflict}}]
+  - ["ĉ", ⠈⠉, {xfail: {backward: conflict}}]
+  - ["Đ", ⠠⠈⠙, {xfail: {backward: conflict}}]
+  - ["Ċ", ⠠⠈⠉, {xfail: {backward: conflict}}]
+  - ["ċ", ⠈⠉, {xfail: {backward: conflict}}]
+  - ["Č", ⠠⠈⠉, {xfail: {backward: conflict}}]
+  - ["č", ⠈⠉, {xfail: {backward: conflict}}]
+  - ["Ď", ⠠⠈⠙, {xfail: {backward: conflict}}]
+  - ["ď", ⠈⠙, {xfail: {backward: conflict}}]
+  - ["ğ", ⠣]
+  - ["ş", ⠩]
+  - ["Ğ", ⠠⠣]
+  - ["Ş", ⠠⠩]
+  - ["ı", ⠔]
+  - ["I", ⠠⠔]
+  - ["i", ⠊]
+  - ["İ", ⠠⠊]
+
+  # sentences
+  - ["(Örnek bir metin)", ⠶⠠⠪⠗⠝⠑⠅ ⠃⠊⠗ ⠍⠑⠞⠊⠝⠶, {xfail: {backward: true}}]
```

Comment:
This test fails after I rebased the branch onto master. In commit 192a04496b375203f51b4bf6655f5f1898776c1e, @jrbowden added the following rule:

    postpunc ) 5-345

The explanation is

> add prepunc and postpunc definitions for several signs to prevent regression backtranslating words followed by punctuation.

So I'm not sure if this was supposed to influence forward translation.

Anyway, I'm not sure what to do with this test.

Reply:
I trying fixing this, readed your review now.
I seed Other @jrbowden doed changes with en-ueb-g1.ctb table.
Anyway, in turkish tables absolute sure need marks 2356 the ( and ) character in grade 1, so perhaps next development cycle better remove en-ueb-g1.ctb table if technical not have a larger need why need include this file by design. :-):-)
Me need syncing the update_turkish_tables branch my local system.

Attila

Reply:
For now I marked it with `xfail`.

Reply:
@hammera I've force-pushed all my changes just now. Please synchronize again.

Reply:
Ok, thanks the notification. I synced again. :-):-)
Understanding what the situation with tr-g1.ctb table level with forward direction, looks following lou_trace example:
```
(alma)
⠶⠁⠇⠍⠁⠐⠜
1.	punctuation	(	2356
2.	lowercase	a	1
3.	lowercase	l	123
4.	lowercase	m	134
5.	lowercase	a	1
6.	postpunc	)	5-345
```
This is easy fixable with following patch (I tested this) now, but not committed yet, because need modifying the ( and ) related test example the grade 1 section in to the 549 line:
```
diff --git a/tables/tr-g1.ctb b/tables/tr-g1.ctb
index b0848ed7..c1ccbc9d 100644
--- a/tables/tr-g1.ctb
+++ b/tables/tr-g1.ctb
@@ -237,7 +237,9 @@ noback always \x25cc 6
 punctuation ' 3
 punctuation ; 23
 punctuation : 25
+prepunc ( 2356
 punctuation ( 2356
+postpunc ) 2356
 punctuation ) 2356
 punctuation − 36
 
diff --git a/tests/braille-specs/tr.yaml b/tests/braille-specs/tr.yaml
index b0548b0a..502b4a44 100644
--- a/tests/braille-specs/tr.yaml
+++ b/tests/braille-specs/tr.yaml
@@ -546,9 +546,7 @@ tests:
   - ["İ", ⠠⠊]
 
   # sentences
-  - ["(Örnek bir metin)", ⠶⠠⠪⠗⠝⠑⠅ ⠃⠊⠗ ⠍⠑⠞⠊⠝⠶,
-     {xfail: {forward: fails since commit 192a04496b (with addition of "postpunc ) 5-345" rule),
-              backward: true}}]
+  - ["(Örnek bir metin)", ⠶⠠⠪⠗⠝⠑⠅ ⠃⠊⠗ ⠍⠑⠞⠊⠝⠶]
   - ["Bu nasıl olabilir?", ⠠⠃⠥ ⠝⠁⠎⠔⠇ ⠕⠇⠁⠃⠊⠇⠊⠗⠦]
   - ["Bu nasıl olabilir…", ⠠⠃⠥ ⠝⠁⠎⠔⠇ ⠕⠇⠁⠃⠊⠇⠊⠗⠲⠲⠲]
   - ["\"En büyük padişah Sultan I. Süleyman'dır.\"",```
```
This fixing way back-translation part work too. :-):-)

And this is the right wanted output the grade 1 forward direction in turkish braille:
```
(alma)
⠶⠁⠇⠍⠁⠶
1.	prepunc	(	2356
2.	lowercase	a	1
3.	lowercase	l	123
4.	lowercase	m	134
5.	lowercase	a	1
6.	postpunc	)	2356
```
So, to winning the precedence rule order the need turkish prepunc and postpunc dot combination, need defining proper rules before en-ueb-g1.ctb include line.
Not beautiful the fix, but works and resolves the unwanted situation with the new en-ueb-g1.ctb table related change doed the turkish grade 1 system. :-):-)
I committing the change.

Attila

Reply:
Done, pushed the change, now running the required checks.
Me the make -j8 check command ran successfull the test directory my local system before pushing.
I drinking a coffee, and perhaps the remaining two online check running right until drinking the cofee too. :-):-)
So, after this change verify result, we ready this PR this development cicle, all situations are resolved.

Attila

Reply:
Nice, all checks passed. :-):-)